modules/http/src/main/scala/higherkindness/mu/http/implicits.scala

```diff
@@ -121,8 +116,18 @@ object implicits {
   def handleResponseError[F[_]: Sync](errorResponse: Response[F]): F[Throwable] =
     errorResponse.bodyAsText.compile.foldMonoid.map(body =>
       ResponseError(errorResponse.status, Some(body).filter(_.nonEmpty)))
+
+  implicit class ThrowableOps(self: Throwable) {
+    def toUnexpected: UnexpectedError =
+      UnexpectedError(self.getClass.getName, Option(self.getMessage))
+  }
+
 }
 
+final case class UnexpectedError(className: String, msg: Option[String])
```

Comment:
The name is a bit generic - most errors are unexpected. The way we're using those, they're closer to "RequestException"s, or 4XX-status `ResponseException`s. Can we combine those error types or rename this one?

Reply:
(resolved offline)